### PR TITLE
Fix being able to buy an item from a vendor even if they dont sell it

### PR DESCRIPTION
**Pull request by EmosewaMC**
Tested that I can still buy items from the commendation vendor in Nexus Tower and from a sentinel vendor in Nimbus Station.

### CODE REVIEW

File: dGame/dGameMessages/GameMessages.cpp [hunk 1]
```diff
@@ -4733,12 +4733,17 @@ void GameMessages::HandleBuyFromVendor(RakNet::BitStream* inStream, Entity* enti
...
 
-	VendorComponent* vend = static_cast<VendorComponent*>(entity->GetComponent(eReplicaComponentType::VENDOR));
+	auto* vend = entity->GetComponent<VendorComponent>();
 	if (!vend && !isCommendationVendor) return;
 
-	InventoryComponent* inv = static_cast<InventoryComponent*>(player->GetComponent(eReplicaComponentType::INVENTORY));
+	auto* inv = player->GetComponent<InventoryComponent>();
 	if (!inv) return;
 
+	if (!isCommendationVendor && !vend->SellsItem(item)) {
```
Comment (aronwk-aaron) [inline dGame/dGameMessages/GameMessages.cpp:4742]: Non-blocking
Should there be a check for the commendation vendor as well? shouldn't be toooooo hard :P
But should probably be it's own PR since it will be a bit more involved

**Review by aronwk-aaron — APPROVED** [review]

File: dGame/dGameMessages/GameMessages.cpp (hunk 1 above)
Comment (EmosewaMC) [reply dGame/dGameMessages/GameMessages.cpp:4742]: He has his own check below.  It checks that the item is actually sold for tokens hence dont check here

**Review by Wincent01 — APPROVED** [review]
